commit 0c03db46c24b3292acf09f8a85a6e5887e69d180 (refs/changes/75/4975/6)
Author: Rui Loura <rui@joyent.com>
Date:   2018-11-27T21:11:08+00:00 (11 months ago)
    
    TRITON-841 Add “Admin IP” and “Admin_IP” to sysinfo output
    Reviewed by: Jorge Schrauwen <sjorge@blackdot.be>
    Reviewed by: Cody Peter Mello <melloc@writev.io>
    Approved by: Cody Peter Mello <melloc@writev.io>

diff --git a/src/sysinfo b/src/sysinfo
index 8f1a3f72..733a059c 100644
--- a/src/sysinfo
+++ b/src/sysinfo
@@ -539,13 +539,25 @@ function get_overlay_nic_tags()
     fi
 }
 
+# Default to Admin_NIC_Tag of "admin"
 function get_admin_nic_tag()
 {
     prop=$(net_conf_grep "^admin_tag=")
     fields=(${prop//=/ })
-    eval "Admin_NIC_Tag=${fields[1]}"
+    eval "Admin_NIC_Tag=${fields[1]:-"admin"}"
 }
 
+function get_admin_ip()
+{
+    # In case get_admin_nic_tag hasn't been called yet.
+    [[ -z "$Admin_NIC_Tag" ]] && get_admin_nic_tag
+
+    # If we are still missing the "Admin_NIC_Tag" default to "admin"
+    tag=${Admin_NIC_Tag:-"admin"}
+    prop=$(net_conf_grep "^${tag}_ip=")
+    fields=(${prop//=/ })
+    eval "Admin_IP=${fields[1]}"
+}
 
 function get_nic_tags()
 {
@@ -696,6 +708,7 @@ Bhyve_Max_Vcpus=${Bhyve_Max_Vcpus}
 HVM_API='${HVM_API}'
 Nic_Tags=${NicTagList}
 Admin_NIC_Tag=${Admin_NIC_Tag}
+Admin_IP=${Admin_IP}
 Setup='${Setup_complete}'
 END
     else
@@ -834,6 +847,7 @@ END
   "CPU Virtualization": "${CPU_Virtualization}",
   "CPU Physical Cores": ${CPU_Count},
   "Admin NIC Tag": "${Admin_NIC_Tag}",
+  "Admin IP": "${Admin_IP}",
 END
     else
         echo "  \"ZFS Quota\": \"${QUOTA}\","
@@ -1026,6 +1040,7 @@ get_hostname
 get_aggregation_mappings
 get_overlay_nic_tags
 get_admin_nic_tag
+get_admin_ip
 get_nic_tags
 get_smartos_network_interfaces
 get_smartos_vnics
